sdk/keyvault/azure-keyvault-certificates/migration_guide.md

```diff
@@ -0,0 +1,288 @@
+
+#### Authenticating
+
+Previously in `azure-keyvault` you could create a `KeyVaultClient` by using `ServicePrincipalCredentials` from `azure-common-credentials`:
+
+```python
+from azure.common.credentials import ServicePrincipalCredentials
+from azure.keyvault import KeyVaultClient
+
+credentials = ServicePrincipalCredentials(
+    client_id="client id",
+    secret="client secret",
+    tenant="tenant id"
+)
+
+certificate_client = KeyVaultClient(credentials)
+```
+
+Now in `azure-keyvault-certificates` you can create a `CertificateClient` with a `DefaultAzureCredential` from [`azure-identity`](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/identity/azure-identity/README.md), where the credential reads the **AZURE_CLIENT_ID** ("client id" above), **AZURE_CLIENT_SECRET** ("client secret" above), and **AZURE_TENANT_ID** ("tenant id" above) environment variables:
+
+```python
+from azure.identity import DefaultAzureCredential
+from azure.keyvault.certificates import CertificateClient
+
+credential = DefaultAzureCredential()
+
+certificate_client = CertificateClient(vault_url="https://my-key-vault.vault.azure.net/", credential=credential)
+```
+
+### Async operations
+
+The modern `azure-keyvault-certificates` library includes a complete async API supported on Python 3.5+. To use it, you must first install an async transport, such as [aiohttp](https://pypi.org/project/aiohttp/). See [azure-core documentation](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md#transport) for more information.
+
+Async operations are available on async clients, which should be closed when they're no longer needed. Each async client is an async context manager and defines an async `close` method. For example:
+
+```python
+from azure.identity.aio import DefaultAzureCredential
+from azure.keyvault.certificates.aio import CertificateClient
+
+credential = DefaultAzureCredential()
+
+# call close when the client is no longer needed
+client = CertificateClient(vault_url="https://my-key-vault.vault.azure.net/", credential=credential)
+...
+await client.close()
+
+# alternatively, use the client as an async context manager
+client = CertificateClient(vault_url="https://my-key-vault.vault.azure.net/", credential=credential)
+async with client:
+  ...
+```
+
+### Create a certificate
+
+In `azure-keyvault` you could create a certificate by using `KeyVaultClient`'s `create_certificate` method, which required a vault endpoint and certificate name. This method returned a `CertificateOperation`.
+
+```python
+from azure.common.credentials import ServicePrincipalCredentials
+from azure.keyvault import KeyVaultClient
+
+credentials = ServicePrincipalCredentials(
```

Comment:
Do we need to include authentication in every example?

Reply:
Probably not, since there's a section about authenticating the client. I'll go ahead and shorten the examples